pkg/auth/requests/token_review.go

```diff
@@ -0,0 +1,78 @@
+package requests
+
+import (
+	"net/http"
+
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	"github.com/rancher/steve/pkg/auth"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	authv1 "k8s.io/client-go/kubernetes/typed/authentication/v1"
+)
+
+var (
+	allowedPaths = map[string]bool{
+		// This is to enable prometheus to use
+		// mounted service account tokens to
+		// fetch metrics.
+		"/metrics": true,
+	}
+)
+
+type TokenReviewAuth struct {
+	AuthClient           authv1.AuthenticationV1Interface
+	InitialAuthenticator auth.Authenticator
```

Comment:
I feel like the InitialAuthenticator call does not belong here. I think we should implement AuthenticatorFunc.Chain(authenticator Authenticator) [in rancher/steve](https://github.com/rancher/steve/blob/e7119828b8f9a9ca2cb2620ba650bd0508ca463e/pkg/auth/filter.go#L39-L43) or something similar to allow us to use something like 

```go
impersonateAuthenticator := requests.NewImpersonatingAuth(sar.NewSubjectAccessReview(clusterManager))
tokenAuthenticator := requests.NewTokenReviewAuth(scaledContext.K8sClient.AuthenticationV1())

authed.Use(mux.MiddlewareFunc(auth.ToMiddleware(impersonateAuthenticator.Chain(tokenAuthenticator))))
```

Reply:
I think the intended way to do this is to use toMiddleware on both and then chain them, I think I should have been doing that anyway since using that function ensure that the user info from the two functions are applied to the response in order. Fixed to do this.